tedge alarm does not transform message fragment to text in the tedge-mapper-c8y (#1715)
thin-edge/thin-edge.io: docs/src/tutorials/raise-alarm.md

An already raised alarm can be cleared by sending an empty message with retained flag enabled to the same alarm topic on which the original alarm was raised.

Topic:
`tedge/alarms/<severity-level>/<alarm-type>`

> Payload:
> ```
> ""
> ```

Reviewer: Technically the payload should be empty, not a literal `""` which this line suggests.

```suggestion
Payload:

*Empty payload*
```

Author: Added an example, I think this is more clearer now.

where the `child-device-id` is the unique device id of the child device.
The alarm payload structure is the same, as described in the previous section.

### Raising an alarm with custom fragment

Thin-edge supports the creation of alarms using custom (user-defined) fragments.
Custom fragments are supported for both the main and child devices.
The custom fragments can be a simple json value or a complex json value.

For example, an alarm with simple custom fragment

Payload:
```json
{
    "text": "Temperature is very high",
    "time": "2021-01-01T05:30:45+00:00",
    "details": "A custom alarm info"
}
```

For example, an alarm with complex custom fragments

Payload:
```json
{
    "text": "Temperature is very high",
    "time": "2021-01-01T05:30:45+00:00",
    "someOtherCustomFragment": {"nested":{"value": "extra info"}}
}
```

> Note: Other than `text` and `time` fields, all the other fields are considered as custom fragments.

### Raising an alarm with empty payload

Alarms can also be raised with a fixed message payload for `thin-edge` as well for the `child` devices.

> The payload will be an `empty json object` message as below

Reviewer 2: Two critical points are missing here. One needs to:

1. Stress the difference between an empty string message payload `""`, that clears the alarm, and an empty json payload `{}` that raises an alarm with default values.
2. Tell what are the default values for the timestamp and the text message (if any) of an alarm created by `{}`.

Author: I have updated the document by adding more details to empty ` the""` message and empty json `{}` message.

Author: If you are mentioning that its for `{}` case then there is already a section here `### Raising an alarm with empty json payload`

where the `child-device-id` is the unique device id of the child device.
The alarm payload structure is the same, as described in the previous section.

### Raising an alarm with custom fragment

Thin-edge supports the creation of alarms using custom (user-defined) fragments.
Custom fragments are supported for both the main and child devices.
The custom fragments can be a simple json value or a complex json value.

> For example, an alarm with simple custom fragment

Author: fixed

Payload:
```json
{
    "text": "Temperature is very high",
    "time": "2021-01-01T05:30:45+00:00",
    "details": "A custom alarm info"
}
```

For example, an alarm with complex custom fragments

Payload:
```json
{
    "text": "Temperature is very high",
    "time": "2021-01-01T05:30:45+00:00",

>     "someOtherCustomFragment": {"nested":{"value": "extra info"}}

Author: fixed

}
```

> Note: Other than `text` and `time` fields, all the other fields are considered as custom fragments.

### Raising an alarm with empty json payload

> Alarms can also be raised with a fixed message payload for `thin-edge` as well for the `child` devices.

Reviewer: An empty payload is used for clearing an alarm (not creating an alarm).

The "fixed message payload" means that the same payload can be used for different alarm types. This is useful for devices what don't care about the payload of the alarm, they just want to indicate that something happened.

> Alarms can also be raised with a fixed message payload for `thin-edge` as well for the `child` devices.

Reviewer 3: Right. Empty payload is the wrong term. But I wish there was a better term to describe an empty JSON payload `{}`, as fixed message payload sounds a bit vague. But, if we're sticking with that term to describe `{}`, then use the same in the section **title** as well.

> Alarms can also be raised with a fixed message payload for `thin-edge` as well for the `child` devices.

Reviewer: Ok, then forget the fixed payload bit (even though the fixed part had nothing to do with the empty json object, just the fact that would mostly like be used when the client can only handle a static/fixed payload for alarms of multiple types).

```suggestion
Alarms can also be raised with an empty json object as the payload.
```

> Alarms can also be raised with a fixed message payload for `thin-edge` as well for the `child` devices.

Author: I agree with @albinsuresh, I was not convinced with the term `fixed message payload` . The `empty json object` suites better.

{
    "text": "Temperature is very high",
    "time": "2021-01-01T05:30:45+00:00",
    "someOtherCustomFragment": {
        "nested": {
            "value": "extra info"
        }
    }
}
```

> Note: Other than `text` and `time` fields, all the other fields are considered as custom fragments.

### Raising an alarm with empty json payload

Alarms can also be raised with an `empty json object` as payload for `thin-edge` as well for the `child` devices.

> The payload will be an `empty json object` message as below

Reviewer 3: ```suggestion
Alarms can also be raised with an `empty json object` payload as follows:
```
Just keeping it short.

Author: fixed

If the alarm is raised from a child device, the payload is published to `c8y/s/us/<child-device-id>` topic instead.

If an alarm contains a `custom fragment` then, the alarm message will be converted to `cumulocity json`
format and then will be published on to `c8y/alarm/alarms/create` topic.

An example of the translated custom message for `thin-edge` device will be as below

```json
{
    "severity":"MAJOR",

>     "type":"pressure_alarm",

Author: Fixed
